app/packages/components/src/components/JSONPanel/JSONPanel.tsx

```diff
@@ -1,35 +1,81 @@
 /**
  * Copyright 2017-2023, Voxel51, Inc.
  */
+import { useState } from "react";
 import { Copy as CopyIcon, Close as CloseIcon } from "@fiftyone/components";
+import CloseRoundedIcon from "@mui/icons-material/CloseRounded";
 import {
-  lookerPanel,
+  lookerPanelV2,
   lookerPanelContainer,
   lookerPanelVerticalContainer,
+  searchContainer,
+  searchInput,
+  copyBtnClass,
+  searchCloseIcon,
 } from "./panel.module.css";
 import {
   lookerCopyJSON,
   lookerCloseJSON,
   lookerJSONPanel,
 } from "./json.module.css";
+import ReactJson from "searchable-react-json-view";
+import { useColorScheme } from "@mui/material";
+
+export default function JSONPanel({ containerRef, onClose, onCopy, json }) {
+  const parsed = JSON.parse(json);
+  const { mode } = useColorScheme();
+  const isDarkMode = mode === "dark";
+  const [searchTerm, setSearchTerm] = useState<string>("");
 
-export default function JSONPanel({ containerRef, jsonHTML, onClose, onCopy }) {
   return (
     <div
       ref={containerRef}
       className={`${lookerJSONPanel} ${lookerPanelContainer}`}
       onClick={(e) => e.stopPropagation()}
     >
       <div className={lookerPanelVerticalContainer}>
-        <div className={lookerPanel}>
-          {jsonHTML && <pre dangerouslySetInnerHTML={jsonHTML} />}
+        <div className={lookerPanelV2}>
+          {parsed && (
+            <div style={{ position: "relative" }}>
+              <div className={searchContainer}>
+                <input
+                  placeholder="Search to highlight text..."
+                  className={searchInput}
+                  onChange={(e) => setSearchTerm(e.target.value)}
+                  value={searchTerm}
+                />
+                {searchTerm && (
+                  <CloseRoundedIcon
+                    className={searchCloseIcon}
+                    onClick={() => setSearchTerm("")}
+                  />
+                )}
+              </div>
+              <ReactJson
+                highlightSearch={searchTerm}
+                src={parsed}
+                theme={`ashes${!isDarkMode ? ":inverted" : ""}`}
```

Comment:
I would be in favor of creating a [custom theme ](https://github.com/mac-s-g/react-json-view#use-your-own-theme)  (perhaps styling, icons, etc. if possible too)

Reply:
agreed that custom theme could be good to have here - That would require some color tuning with different input examples / could take some time. we can add later if no objection

Reply:
tracking it here https://voxel51.atlassian.net/browse/TEAMS-925